library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -1000,15 +1048,33 @@ return_status libspdm_append_message_mut_b(IN void *context, IN void *message,
     return append_managed_buffer(&spdm_context->transcript.message_mut_b,
                      message, message_size);
 #else
-    if (spdm_context->transcript.digest_context_mut_m1m2 == NULL) {
-        spdm_context->transcript.digest_context_mut_m1m2 = spdm_hash_new (
-            spdm_context->connection_info.algorithm.base_hash_algo);
-        spdm_hash_init (spdm_context->connection_info.algorithm.base_hash_algo,
-            spdm_context->transcript.digest_context_mut_m1m2);
+    {
+        boolean result;
+
+        if (spdm_context->transcript.digest_context_mut_m1m2 == NULL) {
+            spdm_context->transcript.digest_context_mut_m1m2 = spdm_hash_new (
+                spdm_context->connection_info.algorithm.base_hash_algo);
+            if (spdm_context->transcript.digest_context_mut_m1m2 == NULL) {
+                return RETURN_DEVICE_ERROR;
+            }
+            result = spdm_hash_init (spdm_context->connection_info.algorithm.base_hash_algo,
+                spdm_context->transcript.digest_context_mut_m1m2);
+            if (!result) {
+                spdm_hash_free (spdm_context->connection_info.algorithm.base_hash_algo,
+                    spdm_context->transcript.digest_context_mut_m1m2);
+                spdm_context->transcript.digest_context_mut_m1m2 = NULL;
+                return RETURN_DEVICE_ERROR;
+            }
```

Comment:
For this and similar functions, the idea is to not initialize the transcript at all, unless the full transcript initialization is successful. Otherwise, we can end up in some unknown state.